chain/types/ethtypes/eth_transactions.go

```diff
@@ -62,9 +62,14 @@ type EthTxArgs struct {
 // - BlockHash
 // - BlockNumber
 // - TransactionIndex
-// - From
 // - Hash
 func EthTxFromSignedEthMessage(smsg *types.SignedMessage) (EthTx, error) {
+	// The from address is always an f410f address, never an ID or other address.
+	if !IsEthAddress(smsg.Message.From) {
```

Comment:
I think, at one point, we thought we were going to support sending "eth transactions" from, e.g., ID addresses. At this point, they can only be sent from f410f addresses, so we can assert this up-front.